server/game/gameSystems/SelectCardSystem.ts

```diff
@@ -73,6 +74,14 @@ export class SelectCardSystem<TContext extends AbilityContext = AbilityContext>
                 ) && properties.cardCondition(card, context);
             properties.selector = CardSelectorFactory.create(Object.assign({}, properties, { cardCondition }));
         }
+
+        if (properties.mode === TargetMode.UpTo || properties.mode === TargetMode.UpToVariable) {
+            properties.canChooseNoCards = properties.canChooseNoCards ?? true;
+        } else {
+            // should not use this property if mode is not UpTo(Variable)
+            properties.canChooseNoCards = false;
```

Comment:
If we shouldn't use this property in certain situations then let's instead have a contract check to throw an error if they set it when they shouldn't. The way it is right now, if someone sets canChooseNoCards to true when they shouldn't then it will just be silently changed to false in a way that they wouldn't realize, which is not good.